src/commands/generate/typeGenerator.ts

```diff
@@ -26,27 +33,67 @@ import {
 import { ImportsVisitor, sortImports } from "./imports";
 import { SimpleAst } from "./simpleAst";
 import { TsReturnTypeVisitor } from "./tsReturnTypeVisitor";
-import { addDeprecatedToDocs, doubleQuote, isValidFunctionName, singleQuote } from "./utils";
+import { ITypeGenerationFlags } from "./typeGenerationFlags";
+import { addDeprecatedToDocs, doubleQuote, isFlavorizable, isValidFunctionName, singleQuote } from "./utils";
 
 export function generateType(
     definition: ITypeDefinition,
     knownTypes: Map<string, ITypeDefinition>,
     simpleAst: SimpleAst,
+    typeGenerationFlags: ITypeGenerationFlags,
 ): Promise<void> {
     if (ITypeDefinition.isAlias(definition)) {
-        // TODO(gracew): implement at some point
-        return Promise.resolve();
+        return generateAlias(definition.alias, knownTypes, simpleAst, typeGenerationFlags);
     } else if (ITypeDefinition.isEnum(definition)) {
         return generateEnum(definition.enum, simpleAst);
     } else if (ITypeDefinition.isObject(definition)) {
-        return generateObject(definition.object, knownTypes, simpleAst);
+        return generateObject(definition.object, knownTypes, simpleAst, typeGenerationFlags);
     } else if (ITypeDefinition.isUnion(definition)) {
-        return generateUnion(definition.union, knownTypes, simpleAst);
+        return generateUnion(definition.union, knownTypes, simpleAst, typeGenerationFlags);
     } else {
         throw Error("unsupported type: " + definition);
     }
 }
 
+/**
+ * Generates a file of the following format:
+ * ```
+ *  export type ExampleAlias = string & {
+ *     __conjure_type?: "ExampleAlias";
+ *     __conjure_package?: "com.palantir.product";
+ *  };
+ * ```
+ */
+const FLAVOR_TYPE_FIELD = "__conjure_type";
+const FLAVOR_PACKAGE_FIELD = "__conjure_package";
+export async function generateAlias(
+    definition: IAliasDefinition,
+    knownTypes: Map<string, ITypeDefinition>,
+    simpleAst: SimpleAst,
+    typeGenerationFlags: ITypeGenerationFlags,
+): Promise<void> {
+    if (isFlavorizable(definition.alias, typeGenerationFlags.flavorizedAliases)) {
+        const tsTypeVisitor = new TsReturnTypeVisitor(knownTypes, definition.typeName, false, typeGenerationFlags);
+        const fieldType = IType.visit(definition.alias, tsTypeVisitor);
+        const sourceFile = simpleAst.createSourceFile(definition.typeName);
+        const typeAlias = sourceFile.addTypeAlias({
+            isExported: true,
+            name: "I" + definition.typeName.name,
```

Comment:
does `definition.typeName.name` need to be pascal-cased or sanitized in any way?

Reply:
it is not done for enum/unions so I assumed it didn't?